routes/verify.go

```diff
@@ -452,3 +484,380 @@ func (fes *APIServer) sendEmail(email *mail.SGMailV3) {
+	idNumber := req.PostFormValue("idNumber")
+
+	// customerId maps to the userReference passed when creating the Jumio session. userReference represents Public Key
+	userReference := req.PostFormValue("customerId")
+
+	// Jumio TransactionID
+	jumioTransactionId := req.PostFormValue("jumioIdScanReference")
+
+	// Get Public key bytes and PKID
+	var publicKeyBytes []byte
+	if userReference != "" {
+		publicKeyBytes, _, err = lib.Base58CheckDecode(userReference)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Problem decoding user public key (customerId): %v", err))
+			return
+		}
+	} else {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Public key (customerId) is required"))
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: error getting utxoview: %v", err))
+		return
+	}
+
+	pkid := utxoView.GetPKIDForPublicKey(publicKeyBytes)
+	if pkid == nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: No PKID found for public key: %v", userReference))
+		return
+	}
+
+	tstamp := uint64(time.Now().UnixNano())
+	// Always log the payload in global state for the PKID so we can search for all jumio verification payloads for a given user.
+	pkidTstampnanosKey := GlobalStateKeyForPKIDTstampnanosToJumioTransaction(pkid.PKID, tstamp)
+	if err = fes.GlobalStatePut(pkidTstampnanosKey, payloadBytes); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Error putting jumio callback in global state: %v", err))
+		return
+	}
+
+	var userMetadata *UserMetadata
+	userMetadata, err = fes.getUserMetadataFromGlobalState(userReference)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Error getting user metadata from global state: %v", err))
+		return
+	}
+
+	// If the user is already jumio verified, this is an error and we shouldn't pay them again.
+	if userMetadata.JumioVerified {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: User already verified: %v", err))
+		return
+	}
+
+	if req.FormValue("idScanStatus") != "SUCCESS" {
+		// This means the verification failed. We've logged the payload in global state above, so we now set JumioReturned
+		// to true and bail.
+		userMetadata.JumioReturned = true
+		if err = fes.putUserMetadataInGlobalState(userMetadata); err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Error putting user metdata in global state: %v", err))
+		}
+		return
+	}
+
+	// Make sure this id hasn't been verified before.
+	uniqueJumioKey := GlobalStateKeyForCountryIDDocumentTypeSubTypeDocumentNumber(idCountry, idType, idSubType, idNumber)
+	// We expect badger to return a key not found error if this document has not been verified before.
+	// If it does not return an error, this is a duplicate, so we skip ahead.
+	if val, _ := fes.GlobalStateGet(uniqueJumioKey); val == nil {
```

Comment:
Do we want to issue a Bad Request if it returns something? Will Jumio continue submitting callbacks if we error?

Reply:
Resubmit behavior is especially important if, for example, balance nanos is too low

Reply:
Jumio will not submit again if we return an error

Reply:
This sounds like a problem then in some error cases but not in others?

Reply:
user can always go through Jumio flow again since we won't mark them as jumio verified. worst case scenario is that we give them money and then something fails when we are putting the user's data in global state - this error should occur very rarely. If we get an error because the balance is too low to cover the starter clout amount, the user can always go through the flow later. We should just monitor the balance that send CLOUT here and make sure it's always got enough juice. Setting up a retry queue for this will be a bit annoying.